src/cascade/model/age_time_grid.py

```diff
@@ -0,0 +1,153 @@
+from datetime import timedelta
+from itertools import product
+from math import nan, inf
+
+import numpy as np
+import pandas as pd
+
+from cascade.dismod.constants import PriorKindEnum
+
+GRID_SNAP_DISTANCE = 1 / timedelta(days=365).total_seconds()
+"""Times within one second are considered equal."""
+
+
+class AgeTimeGrid:
+    """The AgeTime grid holds rows of a table at each age and time value.
+
+    At each age and time point is a dataframe consisting of the columns
+    given in the constructor. So getting an item returns a dataframe
+    with those columns. Setting a dataframe sets those columns.
+    Each AgeTimeGrid has three possible mulstds, for value, dage, dtime.
+
+    >>> atg = AgeTimeGrid(([0, 10, 20], [1990, 2000, 2010]), ["height", "weight"])
+    >>> atg[:, :] = [6.1, 195]
+    >>> atg[:, :].height = [5.9]
+    >>> atg[10, 2000] = [5.7, 180]
+    >>> atg[5:17, 1980:1990].weight = 125
+    >>> assert (atg[20, 2000].weight == 195).all()
+    >>> assert isinstance(atg[0, 1990], pd.DataFrame)
+
+    If the column has the same name as a function (mean), then access it
```

Comment:
Also, is this even true? `atg[:, :]` will always return a DataFrame (I think) and indexing a method name out of a dataframe doesn't work this way.

Reply:
```python
>>> df = pd.DataFrame(dict(a=[1,2,3], mean=[4,5,6]))
>>> df.a
0    1
1    2
2    3
Name: a, dtype: int64
>>> df.mean
<bound method DataFrame.mean of    a  mean
0  1     4
1  2     5
2  3     6>
>>> df["mean"]
0    4
1    5
2    6
Name: mean, dtype: int64
```
I don't like this. I'm torn between using "mean" because it's everywhere in Dismod's documentation and because, for Priors, it's actually the correct term, and using another name. Next guess is "value", but that's like the "values" property.

Reply:
Oh... When you say "has the same name as a function" you don't mean that it will execute that function but that you will populate the data with the results of that function under the expected name? That's less weird than what I thought. But is that happening somewhere? I don't see it. Or am I just confused?

Reply:
It's not going wrong anywhere. It's a very common pitfall in Pandas, so I mentioned it in the user docs. I don't know whether it's appropriate to add that as an aside, but it's a guess.

Reply:
Hmm. Actually rereading what you wrote, you clearly didn't mean what I thought you meant. You are describing what to do not describing a behavior that the system will have... I'm not sure. It wasn't clear to me but that is probably my fault.